lib/puppet/parser/functions/count.rb

```diff
@@ -4,8 +4,20 @@
 module Puppet::Parser::Functions
   newfunction(:count, :type => :rvalue, :arity => -2, :doc => <<-DOC
     Takes an array as first argument and an optional second argument.
-    Count the number of elements in array that matches second argument.
-    If called with only an array it counts the number of elements that are not nil/undef.
+    Count the number of elements in array that is equal to the second argument.
+    If called with only an array it counts the number of elements that are not nil/undef/empty-string.
+
+    Note: equality is tested with a Ruby method and it is therefore subject to what Ruby considers
+    to be equal. For strings this means that equality is case sensitive.
+
+    In Puppet core, counting can be done in general by using a combination of the core functions
```

Comment:
The 'p' in Puppet should be uppercase.

Reply:
Hm, the line in question has `Puppet` with capital `P` already ???